### PR TITLE
Suppressor makes bullets less visible

### PR DESCRIPTION
**Pull request by ICE-IS-NICE**
## About The Pull Request
This PR gives suppressor a unique mechanic of making fired bullets a bit less visible. Before - After.
readydefaultready64
## Why It's Good For The Game
A suppressor does hide a muzzle flash and bullets don't emit light, but it's almost useless, because they are still obviously visible and it makes almost no noticable difference. It's still like firing tracer bullets. A suppressor doesn't fully represent "spec ops" or "tacticool" (whatever you name it) vibes and it's a bit dissapointing. It has some good points but truly isn't the best choice at anything (either accuracy, recoil or scatter, not even mentioning its noticable downsides). This PR grants a suppressor truly unique feature, perhaps giving marines a bit more subtle way of playing (not just run n' gun).
You may call it "suppressor buff" but I'd insist on calling it "uncovering its potential". Suppressor should be about making a gun more "subtle", more "delicate", more "finesse". Thus a suppressor will become a truly unique attachement, not just "ordinary extended barrel alternative".
## Changelog
:cl:
add: Suppressor makes bullets less visible.
/:cl:

### CODE REVIEW

**Comment by ivanmixo** [comment]
Very against this, suppressor is already arguably problematic due to it hiding what gun you're being shot by (since sound is your main indicator). You **need** to know what gun you're being shot by in order to gauge how long you can stay in a fight, this would make it even more problematic because not only would you not be able to tell what is shooting you, you also wouldnt be able to tell from where (mechanics which rely on your eyesight being good are cancer by the way) 

Tl;dr silencer removes very important feedback which xenos rely on to be able to properly take fights, and this would make that problem even worse.

**Comment by spookydonut** [comment]
what was the chatgpt prompt you used?

**Comment by ivanmixo** [comment]
> what was the chatgpt prompt you used?

I am going to commit violence

**Comment by SimplySmiley123** [comment]
> Very against this, suppressor is already arguably problematic due to it hiding what gun you're being shot by (since sound is your main indicator). You **need** to know what gun you're being shot by in order to gauge how long you can stay in a fight, this would make it even more problematic because not only would you not be able to tell what is shooting you, you also wouldnt be able to tell from where (mechanics which rely on your eyesight being good are cancer by the way)
> 
> Tl;dr silencer removes very important feedback which xenos rely on to be able to properly take fights, and this would make that problem even worse.

Problem with the former is that it's only really an issue in a 1v1 between the xenos and the marine, at which point, lol, lmao, how is the marine in this case alive anyway.
And at longer ranges, you'll prolly get hit once, same as if someone shot you without the suppressor, no real difference here atm.

Since in a team fight, there's gonna be a cacophony of various guns, screams, xeno screeches, etc. alongside hundreds of bullets from the general direction marines are at to the point of borderline sensory overload.
Silencer barely ever matters.

**Comment by ivanmixo** [comment]
Every single thing you say is a huge assumption, you can definitely pick out dangerous guns from all the noise happening even in big fights.

**Comment by Mantlecrawler** [comment]
I think this is fine; as a beno main what matters is wound overlays, Stat panel, and health hud, not what bullets I see or hear

**Comment by Isy232** [comment]
This is a significant suppressor buff but the suppressor is so useless as is that I fail to see the issue.
The suppressor halves bullet speed and means you cant take an extended barrel, which is likely to significantly decrease accuracy at medium to long range.

Ivan brings up a decent point but I think the decreased bullet velocity will make the scary gun a lot less effective at hitting retreating xenos.

**Comment by Pariah919** [comment]
This might make suppressor obnoxious to play against with snipers, but that's generally a level of pain and bullet travel time few people want to deal with. Though the entire purpose of the attachment is to already hide your gunfire and what you're firing.

**Comment by Lumipharon** [comment]
> This is a significant suppressor buff but the suppressor is so useless as is that I fail to see the issue. The suppressor halves bullet speed and means you cant take an extended barrel, which is likely to significantly decrease accuracy at medium to long range.
> 
> (swapping from extended barrel to suppressor is a 4x decrease in bullet velocity, to my knowledge)
> 
> Ivan brings up a decent point but I think the decreased bullet velocity will make the scary gun a lot less effective at hitting retreating xenos.

This isn't how these attachments work at all.
Most, but not all, projectiles travel 3 tiles per tick. Some like snipers move at 4 per tick. 

Suppressor reduces it by 1. They also slightly increase damage falloff.
Extended barrel increases it by 1.
Barrel charger increases it by 2. Also slightly reduces damage falloff.

Being less effective at range is kind of a moot point since that's not what it's meant for.

**Review by TiviPlus — APPROVED** [review]
code ok @Kuro020
